src/Symfony/Component/Workflow/MarkingStore/MarkingStoreInterface.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Workflow\MarkingStore;
+
+use Symfony\Component\Workflow\Marking;
+
+/**
+ * MarkingStoreInterface.
+ *
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+interface MarkingStoreInterface
+{
+    public function getMarking($subject);
```

Comment:
IDEs cannot guess the return type of an interface, as there is no code in the interface to guess it.

And IDEs guessing it based on implementations would be broken IDEs, as they cannot be sure that the implementation returns all possible return value types (for instance, the return type could be an interface rather than an implementation)